src/ledger/LedgerManagerImpl.cpp

```diff
@@ -619,9 +624,23 @@ LedgerManagerImpl::startCatchup(CatchupConfiguration configuration,
         throw std::invalid_argument("Target ledger is not newer than LCL");
     }
 
-    setCatchupState(CatchupState::APPLYING_HISTORY);
-    assert(manualCatchup == mSyncingLedgers.empty());
+    if (manualCatchup)
+    {
+        if (getState() == LM_CATCHING_UP_STATE &&
+            mCatchupState != CatchupState::WAITING_FOR_CLOSING_LEDGER)
+        {
+            throw std::runtime_error(
+                "Unable to perform second catchup before first one finishes");
+        }
+        assert(mSyncingLedgers.empty());
+        mSyncingLedgers.reset();
```

Comment:
Is this the right place to reset `mSyncingLedgers`? I think it should be done when catchup is finished (which already happens in `finalizeCatchup`).

Reply:
Also, wrt asserts, with the new explicit reset, I don't think checking `mSyncingLedgers.empty()` is enough anymore , so you might want to make it stricter and check `mLastLedgerSeq` as well (maybe by introducing a new method `bool SyncingLedgerChain::isReset()`)

Reply:
This place for reset is required by "manual" catchup. I don't like it. I don't like manual catchup from CommandHandler at all (it causes a lot of trouble, and is probably not used anywhere).

I would like to replace manual notion with "Online" and "Offline", where Online requires going throught WAITING_FOR_CLOSING_LEDGER state and Offline does not.

But anyway, this one line is for when you are doing manual catchup from CommandHandler and then another one later. And mSyncingLedgers must be empty here, because we are in WAITING_FOR_CLOSING_LEDGER, which is set when mSyncingLedgers is empty, and adding anything to mSyncingLedgers requires being not in WAITING_FOR_CLOSING_LEDGER state.